### PR TITLE
Collectibles :D

### PR DESCRIPTION
**Pull request by Just-Jojo**
## Type
- [ ] Bugfix
- [ ] Enhancement
- [ ] New Feature
- [x] New Cog

## Description of the changes
Add the Collectible cog

## Todo list
- [x] Add/remove Collectibles
- [x] Make sure that currency doesn't get devoured on accident
- [x] Other things(?) eg. command and embed touch-ups, etc.

### CODE REVIEW

**Comment by Just-Jojo** [comment]
This is ready to be removed :D

**Comment by Just-Jojo** [comment]
Reviewed not removed, curse me keyboard

File: collectibles/collectibles.py [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+log.setLevel(logging.DEBUG)
+_config_structure = {
+    "user": {
+        "collectibles": {},
+        "total_count": 0,
+    },
+    "guild": {
+        "collectibles": {},
+    },
+    "global": {"collectibles": {"Trophy": ["🏆", 300]}},
```
Comment (Kreusada) [inline collectibles/collectibles.py:26]:
```suggestion
    "global": {"collectibles": {"Trophy": ["\N{TROPHY}\N{VARIATION SELECTOR-16}", 300]}},
```
Lets change this to unicode

File: collectibles/collectibles.py [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+    __author__ = ["Jojo#7791"]
+    __version__ = "1.0.0"
+
+    def __init__(self, bot: Red):
+        self.bot = bot
+        self.config = Config.get_conf(self, 544974305445019651, True)
+        for key, value in _config_structure.items():
+            getattr(self.config, f"register_{key}")(
+                **value
+            )  # am too lazy to type it all out :)
```
Comment (Kreusada) [inline collectibles/collectibles.py:42]: I mean this works but I dunno if Red QA will like it...

File: collectibles/collectibles.py [hunk 3]
```diff
@@ -0,0 +1,223 @@
...
+            )
+        msg = msg[:-2] if msg.endswith("\n") else msg
+        await Menu(source=Page(list(pagify(msg)), "Collectibles")).start(ctx)
+
+    @collectible.command(
+        name="globaladd",
+        aliases=[
+            "gadd",
+        ],
+    )
```
Comment (Kreusada) [inline collectibles/collectibles.py:137]:
```suggestion
    @collectible.command(name="globaladd", aliases=["gadd"])
```
Personally I don't like the styling changes here

**Review by Kreusada — APPROVED** [review]
LGTM, few styling changes that's all

File: collectibles/collectibles.py (hunk 2 above)
Comment (Just-Jojo) [reply collectibles/collectibles.py:42]: That's fair lol

**Comment by Just-Jojo** [comment]
Thanks!

**Pull request merged**